# 第一节课知识点小结
![第一节课知识点小结](./picture/2-1.jpg)
大家是否还记得《美国队长2》剧情中的“洞察计划”：通过一个人过去的行为，包括说过的话，看过的书，听过的音乐，看过的电影，参加的社团，发表、点赞、转发的微博，发送的电子邮件，每一通电话、每一条短信，活动圈子，职业属性，教育经历，医疗情况，血型星座等，来对这个人的未来进行预测。    
这些电影中收集的信息、概念都可在Python江湖中对应。比如：电子邮件中的文字内容，点赞微博的数字量、血型星座中的表情符号在Python江湖中就是数据。    
数据五花八门，各式各样，颜面都不一样。只有记清它们的颜面，了解它们的擅长专属， 才能一起探究美国队长的洞察计划    
![python中的数据](./picture/2-2.jpg)
今天要讲的主要内容就是美国队长洞察计划中的数据，完成与AI的对话。让小K带着大家一起探究美国队长的洞察计划。    
# 美国队长的洞察计划
## 数据类型
在Python江湖里，常用的数据类型有三种，字符串，整数，浮点数    
### 字符串
首先， 我们来看一下“黄袍加身”的字符串， 字符串英文名string，简称str。 其实， 在上一关初入江湖的时候， 你已经用过它了，看面庞很熟悉。    
代码片段    
```
name='哪吒'
print("Let's go go go")
```
是不是很熟悉? 你猜对了，穿上单引号、双引号、三引号黄袍的内容就是字符串，无论引号里面内容是中文、英文、法文、数字、符号、甚至是火星文。
在上面代码里 '哪吒' ，"Let's go go go" 都属于字符串类型。    
下面, 请观察一下左边代码, 然后点击运行按钮, 跟小K 一起体验字符串的应用。    



In [3]:
movie = '美国队长2'
name = 'The Winter Soldier'
price = "7.14"
word = '''≡(▔﹏▔)≡'''
print(movie)
print(name)
print(price)
print(word)

美国队长2
The Winter Soldier
7.14
≡(▔﹏▔)≡


感觉怎么样? 字符串是不是很简单。    
字符串, 江湖行话, 就是由一个个字符连接起来的组合, 你平时所见的字母、数字、汉字、符号都是字符。 字符串可以用来表示词、语句、甚至是数学公式。    
下面，小K带你去探究美国队长洞察计划中字符串的用法。请你来试一试，输出下面代码框中的内容，具体文本可以直接复制。【江湖秘籍：记得用引号括住，创建字符串；将字符串赋值给变量；使用print（）函数打印。】    
我们强者就该无所畏惧。    
那你为什么要跑？    

In [5]:
secret = '''
我们强者就该无所畏惧。
那你为什么要跑？
'''
print(secret)


我们强者就该无所畏惧。
那你为什么要跑？



In [7]:
say = '我们强者就该无所畏惧。'
print(say)
say1 = '那你为什么要跑？'
print(say1)

我们强者就该无所畏惧。
那你为什么要跑？


江湖秘籍：字符串类型必须有引号。字符串没有引号，就会向你吐槽的。    
下面，我们来总结一下字符串小结。  
![字符串小结](./picture/2-3.png)
### 整数
是时候，脱去黄马甲了，某些数据脱去黄马甲， 也照样能打印出来。比如上一关的print(1314)    
这就是我们常见的数据类型第二种：整数， 英文为integer， 简称int。 Python江湖的整数和我们现实数学中定义的一样：整数是正整数、零和负整数的统称。说白了，就是没有小数点的数字。    
在下面代码中，所有变量的内容都是整数。    
代码片段
```
age = 18 
height = 192 

weight = 80
```
选择题
现在，我们趁热打铁，请问以下三个数据 “hello world” 36 “72” 的数据类型分别为：    
A：字符串 整数 字符串    
B：字符串 整数 整数    
C：文字 数字 英文    
恭喜你答对了，选A。 由引号括起来的“hello world” “72”是字符串，无引号的36才是整数。    
整数就是一个不带小数点的数字，还不能穿黄马甲。直接用print()函数打印。 比如如下代码:    
编程题    

In [9]:
# 打印整数
print(521)

521


整数和字符串不一样，它不能穿黄马甲，只能独自玩耍，一旦遇到其他文字、字母、符号立马会吐槽。    
就像运行下面代码这样会报错：NameError: name '老铁666' is not defined    
编程题

In [11]:
num="老铁666"
print(num)

老铁666


![整数知识点](./picture/2-4.jpg)
### 浮点数
整数是不带小数点的纯数字，那么带小数点的纯数字怎么表示呢？Python江湖中，将带有小数点的纯数字定义为浮点数。 它是我们从常见的数据类型之一。    
浮点数是显而易见的，它比整数多了一个小数点【.】，下列代码中的数字都是浮点数。    
代码片段    
```
3.1415926
95.27
999.00
```
浮点数，英文称为float。只是浮点型没有简写形式。   
浮点数虽说比整数多一个小数点，但是它存在一个bug。因为浮点数的特点，导致它在计算机内运算时，结果不是那么精确，会有四舍五入的差。这是什么意思呢？    
下面，请你先口算一下【0.35+0.11】的结果？    
然后我们使用Python程序运行一下， 请点击下面运行按钮，并阅读答案：    
编程题    

In [14]:
print(0.35+0.11)

0.45999999999999996


呀？ 怎么不是0.46呢，而是一个比0.46小的数字？ 这是因为Python计算浮点数时，会把0.35与0.11转换成二进制数 【江湖秘籍：二进制数由0和1表示，逢2进1】    
### 二进制转换
```
0.35（十进制） = 0.010110011001100110011001100110011001100110011001100111（二进制）
0.11（十进制）= 0.00011100001010001111010111000010100011110101110000101001（二进制）
```
在这个十进制转换二进制过程中，产生了误差，这样就造成了我们与结果的误差。    
然后，这两个二进制数字相加后，在将得到的二进制结果转换成十进制小数。    
不过对于浮点数， 你不要方，在前期我们很少会涉及浮点数运算。在后期的基础课，小K带你接触更多的相关知识，加深对浮点数的理解。    
下面我们来回顾一下浮点数的主要内容。    
![浮点数知识点](./picture/2-5.jpg)
![数据类型](./picture/2-6.jpg)
选择题    
请问以下数据【88】、【-3】、【‘99’】的数据类型分别为：    
A：整数、小数、字符串    
B：整数、整数、字符串    
C：整数、浮点数、整数    
恭喜你答对了，选B。88是正整数属于整数，-3是负整数属于整数 '99'有引号属于字符串。    
### 四则运算
Python江湖中的整数、浮点数和数学上的整数、小数是一样的，它最重要的作用是结合运算符号进行数学计算。 接下来我们看一下Python如何利用整数、浮点数进行四则运算。    
Python江湖中的四则运算，法则和数学上的一致，符号有点差别。    
首先，我们看一看Python江湖的运算符    
![Python算数运算符](./picture/2-7.jpg)
还是加减乘除，但是乘与除的符号变了哦。    
江湖秘籍：先把这张图存下来，放到储物袋里，等需要用到放大招的时候，再拿出来看看。    
有了四则运算，再说一下Python江湖里的运算优先级。    
啥叫优先级呢？    
先算乘除，再算加减，有括号的先算括号里的，从左往右。这不和数学上的运算一样一样一样的嘛。    
![简单运算优先级口诀](./picture/2-8.jpg)

In [16]:
print(999.99*56-103*50)
print((55+22)*99+(38-21)*10)

50849.44
7793


我们看一下print((55+22)*99+(38-21)*10)的运算顺序：    
第一步计算55+22 =77; 第二步计算77*99= 7623; 第三步计算38-21=17; 第四步计算17*10 =170 ; 第五步计算7623+170 = 7793 。   
好复杂的运算啊，算的有点慢。算的有点慢，就把这个工作交给Python干呀。省下的时间就可以风花雪月了。    
选择题    
那么，按照这个运算法则，请问以下代码的计算顺序是什么？print（（55-22）*3 -99/9）    
A： （55-22） 上一步结果*3 上一步结果-99上一步结果除9    
B：（55-22） 上一步结果*3乘法结果-99/9除法结果    
C：(55-22） 3-99 上一步结果/9 减法的结果*除法的结果    
恭喜你答对了，选B。 按照运算法则，先算括号，再算乘除，最后加减，自左至右。    
好了，我们已经在掌握了洞察计划中的四则运算了。下面，我们来探究一下，如何在Python江湖中将不同的词语内容组成一句话。   
字符串的拼接    
例如，小K这里有几个零碎的词语，也就是变量。然后组合成一句话，‘美国队长取材于漫威漫画‘，在Python江湖中，你要如何实现呢？    
代码片段    
```
hero = '美国队长'
title = '漫威漫画'
action = '来自于' 
```
#输出 美国队长来自于漫威漫画    
这就涉及到了我们下一个知识点—字符串拼接。    
字符串拼接的方法简单的很， 就是利用字符串拼接符号【+】，将需要拼接的变量连接在一起。    
比如上面的例子，我可以如何写。    

In [18]:
hero = '美国队长'
title = '漫威漫画'
action = '取材于'
print(hero+action+title)

美国队长取材于漫威漫画


有了拼接符号, 我们就可以拼接我们想要的内容.’ 美国队长取材于漫威漫画’. 当然我们还可以调整拼接顺序, 输出不同内容.    
下面我们就来试一下，如何打印出 '美国队长于漫威漫画编写'    

In [20]:
hero = '美国队长'
title = '漫威漫画'
action = '编写'
place= '于'
print(hero+place+title+action)

美国队长于漫威漫画编写


学会了字符串拼接,就能让我们整合数据更加灵活, 组合出我们想要的任意内容。    
这就是我们所学的字符串拼接。    
![数据拼接](./picture/2-9.jpg)
好了，现在你已经是探究洞察计划里面的大神了。 那么如何在Python江湖中玩的更6呢。 下面我们就要学会施展法术，输出 '美国队长2'。    


In [22]:
name = '美国队长'
num = '2'
print(name+num)

美国队长2


理想很丰满，现实很骨感。 又向我们吐槽了。 江湖秘籍:只有字符串与字符串才能拼接。   
到底是那里不对呢？你会发现num=2中的2没有引号，也就是说num不是字符串，而是一个整数。    
就是它，出不来结果，就是num的锅。    
那么问题来了，当我们探查洞察计划中更多的数据时怎么办？ 总不能通过肉眼一个一个去看呀。    
这一点，Python已经给我写好了，它抛给我们一个月光宝盒。 能够帮助我们来查询不确定数据的类型。 这个月光宝盒是一个函数，叫做type()函数 。
数据类型的查询—type()函数    
这个type()函数应该如何使用呢？ So easy，只需要把查询的内容放进括号里就得了。    
![type()函数](./picture/2-10.jpg)
只要你把内容写在括号里，type()函数会自动查询它的类型。    

In [24]:
name = '美国队长'
num = 2
print(type(name))
print(type(num))

<class 'str'>
<class 'int'>


name = '美国队长' 查询的结果是< class 'str'>, 属于字符串类型。    
num = 2 查询的结果是< class 'int'>, 属于整数类型。    
在月光宝盒type()函数的帮助下, 我们可以一一识别洞察计划中各类数据的类型。    
貌似到现在, 我们的问题还没有解决啊, 我们还是没有将name = '美国队长'与num = 2拼成'美国队长2'啊.    
为什么一个字符串类型的变量与一个数字类型的变量不能连接起来呢? 就是他俩不是一类人。    
打个比方，美女配英雄。 总不能美女配野兽啊。 美女和野兽聊不起天啊。    
但是, 你现在非得让美女配野兽, 怎么办? 怎么才能把两个不同类型的数据拼在一起?    
这就涉及到了这堂课的最后一个知识点: 数据转换。    
### 数据转换
数据转换的方法有3个: str() ，int()，float()    
下面我们先来看一下str()函数。    
### Str()函数
Str()函数可以帮助我们解决刚才问题，它可以将其他类型的数据变换成str类型。    
![数据类型转换方法](./picture/2-11.jpg)
Str()函数， 不管你之前属于啥类型。 只要到了我的地盘，就得听我的，我就把任何数据都转成字符串类型。    
下面，运行一下左边代码，看如何完成字符串拼接。    

In [26]:
name = '美国队长'
num = 2
print(name+str(num))

美国队长2


选择题    
请问, 下面选项的代码, 运行后, 哪一句会报错？    
A：print(9+6+5)    
B：print('Python'+'江湖')    
C：print('小女今年'+18)    
恭喜你答对了，选C。18是int类型，与字符串类型拼接肯定报错。    
下面, 我们总结一下str()函数    
![数据类型转换方法](./picture/2-12.jpg)
既然整数类型能够转换成字符串类型,那字符串类型能转换为整数吗?    
那必须的啊。    
### int()函数
![int函数转换](./picture/2-13.jpg)
int()函数可以将其他类型的数据转换成整数类型，像str()一样，你只需要将转换的数据放进int()中。    
下面，我们来试验一下，请补全代码，计算num1与num2的和。    


In [27]:
num1 = '3'
num2 = '6'

print(int(num1))
print(int(num2))

3
6


不过对于int()函数的使用，大家要注意一点: 只有字符串中的内容是纯数字时，才能够使用int()函数进行强制转换。    
对于文字、英文、火星文都不能使用int（）进行转换。    
浮点数形式的字符串也是不能使用int()强制转换的。    
比如下方代码，如果点击运行就会报错。【ValueError: invalid literal for int() with base 10: '9.9'】    
江湖秘籍: 浮点类型的字符串是无法使用int()进行数据转换的。    
代码片段
```
print(int('9.9'))
```
运行后结果：ValueError：invalid literal for int() with base 10: '9.9'    
这是不是就说浮点数不能转换成整数类型了?    
不, 虽然浮点形式的字符串不能使用int()函数转换, 但是浮点数是可以使用int()函数来转换的。   
选择题    
请找出下列选项中, int()不能转换为整数的数据 。    
A：‘2019’    
B：99.99    
C：’38.56’    
恭喜你答对了，选C。 C项 '38.56'是字符串形式的浮点数，int()会报错的。    
最后, 小结一下int()函数使用    
![int()函数使用小结](./picture/2-14.jpg)
可是, 如果我想把字符串形式的浮点数’38.56’转换成浮点数怎么办?    
有法子, 这就涉及到我们最后一个知识点—float()函数 。    
float()函数    
![float()函数](./picture/2-15.jpg)
float()函数的使用和str()、int()的方法一致。也是将需要转换的数据放在括号里。    
float()函数可以将整数和字符串转换成浮点数类型，但前提是，这个字符串中的内容一定是数字形式。    
经过str()与int()的探究, float()是不是更好懂一些?    
下面, 让我们敲段代码来试试， 请补齐代码， 将下面所有变量转换为浮点类型,并打印。    

In [28]:
height = 198.2
weight = 97
age = '30'

print(float(height))
print(float(weight))
print(float(age))

198.2
97.0
30.0


![数据转换总结](./picture/2-16.jpg)
怎么样? 掌握了数据类型、数据拼接、数据转换的你, 已经可以加入到美国队长的团队, 一起探究洞察计划中的数据了。    
最后,让我们总结一下这关你都学到了什么   
![总结](./picture/2-17.jpg)  
## 练习
题目要求    
请运用所给变量，使用str()函数打印两句话。    
第一句话：1囚犯：你有什么能力来带我们出去?    
第二句话：2 CA：我揍了希特勒200多次。    
其中，变量会在【书写代码】提供，请直接【复制粘贴】：    
题目讲解    
在Python的江湖中，最常用的数据类型有三种：字符串(str)、整数(int)和浮点数(float)。    
在数据拼接中，为了将不同的信息进行整合，可以使用拼接符号。但是，如果数据非字符串类型，则无法进行拼接。    
此时，我们可以使用数据转换函数str()，将数据转换为字符串类型后，再进行拼接。    
书写代码    

In [30]:
num1  = 1
num2  = 2
name1 = "囚犯"
name2 = "CA"
word1 = "你有什么能力来带我们出去?"
word2 = "我揍了希特勒200多次。"
fuhao1 = ":"

print(str(num1) + name1 + fuhao1 + word1)
print(str(num2) + name2 + fuhao1 + word2)

1囚犯:你有什么能力来带我们出去?
2CA:我揍了希特勒200多次。


### 题目要求
请运用所给变量，使用数据转换str()、int()、float()及数据拼接符号+,    
打印一句话：美国队长2该片于2014年4月4日在北美与中国同步上映。    
其中，变量会在【书写代码】提供，请直接【复制粘贴】：    
题目讲解    
在数据拼接中，为了将不同的信息进行整合，可以使用拼接符号。但是，如果数据非字符串类型，则无法进行拼接。此时，我们可以使用数据转换函数str()，将数据转换为字符串类型后，再进行拼接。并且针对不同类型的数据，我们需要经历多次转换，最后才能转为字符串类型。（注意，题干给的数字是2.5哦，想想怎么把2.5转化成2））    


In [31]:
name = "美国队长"
num = 2
word = "该片于2014年4月4日在北美与中国同步上映。"

print(name + str(num) + word)

美国队长2该片于2014年4月4日在北美与中国同步上映。


In [1]:
#列表
transformers = ['李雷', '韩梅梅', '180', '3.5']
print(transformers)
print(type(transformers))

transformers = ['擎天柱','大黄蜂','救护车']
print(transformers[1])

transformers = ['擎天柱','大黄蜂','救护车','巨无霸福特','红蜘蛛']

#左闭右开区间
print(transformers[:])

print(transformers[2:])

print(transformers[:2])

print(transformers[1:3])

print(transformers[2:4])

names = ['李雷','lily','lucy','韩梅梅']
print(names[2])
print(names[1:3])

transformers = ['擎天柱','大黄蜂','救护车','巨无霸福特','红蜘蛛']

transformers.append('萨克巨人')

print(transformers)

transformers = ['擎天柱','大黄蜂','救护车','巨无霸福特','红蜘蛛']

del transformers[2]

print(transformers)

transformers = ['擎天柱','大黄蜂','救护车','巨无霸福特','红蜘蛛']

fc = { '擎天柱': 95 ,'大黄蜂':90 ,'救护车':86, '巨无霸福特':80,'红蜘蛛':80  }

print(len(transformers))

print(len(fc))

print(type(transformers))

print(type(fc))

fc = { '擎天柱': 95 ,'大黄蜂':90 ,'救护车':86, '巨无霸福特':80,'红蜘蛛':80  }

print(fc['擎天柱'])

fc = { '擎天柱': 95 ,'大黄蜂':90 ,'救护车':86, '巨无霸福特':80,'红蜘蛛':80  }

del fc['巨无霸福特']

print(fc)

fc['巨无霸福特'] = 50

print(fc)

['李雷', '韩梅梅', '180', '3.5']
<class 'list'>
大黄蜂
['擎天柱', '大黄蜂', '救护车', '巨无霸福特', '红蜘蛛']
['救护车', '巨无霸福特', '红蜘蛛']
['擎天柱', '大黄蜂']
['大黄蜂', '救护车']
['救护车', '巨无霸福特']
lucy
['lily', 'lucy']
['擎天柱', '大黄蜂', '救护车', '巨无霸福特', '红蜘蛛', '萨克巨人']
['擎天柱', '大黄蜂', '巨无霸福特', '红蜘蛛']
5
5
<class 'list'>
<class 'dict'>
95
{'擎天柱': 95, '大黄蜂': 90, '救护车': 86, '红蜘蛛': 80}
{'擎天柱': 95, '大黄蜂': 90, '救护车': 86, '红蜘蛛': 80, '巨无霸福特': 50}


In [20]:
transformers = ['猛大帅' , '铁甲龙','六面兽', '大力金刚', '救护车','红色警戒', '大无畏', '飞天虎','闪电','冲云霄', '计算王','求雨鬼', '守护神','混天豹', '擎天柱', '威震天','惊破天', '红蜘蛛', '巨无霸福特', '雷霆解救队' ]
for i in transformers:
    print(i+'睡了没？')
    
for i in [1,2,3,4,5,6]:
    print(str(i)+'取钱')
print(i)

city  = {'北京':'天安门','上海':'东方明珠','广州':'珠江'}

for i in city:
    print(i)
    
name = '黄晓明'
for i in name:
    print(i)
    


猛大帅睡了没？
铁甲龙睡了没？
六面兽睡了没？
大力金刚睡了没？
救护车睡了没？
红色警戒睡了没？
大无畏睡了没？
飞天虎睡了没？
闪电睡了没？
冲云霄睡了没？
计算王睡了没？
求雨鬼睡了没？
守护神睡了没？
混天豹睡了没？
擎天柱睡了没？
威震天睡了没？
惊破天睡了没？
红蜘蛛睡了没？
巨无霸福特睡了没？
雷霆解救队睡了没？
1取钱
2取钱
3取钱
4取钱
5取钱
6取钱
6
北京
上海
广州
黄
晓
明
0
1
2
3
4


In [33]:
# 请直接运行代码
for i in range(5):
    print(i)
    
# 请直接运行代码
for i in range(11,15):
    print(i)
    
# 请直接运行代码
for i in range(11,100,2):
    print(i)
city  = {'北京':'天安门','上海':'东方明珠','广州':'珠江'}
for i in city:
    print(city[i])
    
    x = 0 
while x < 6:
    x = x+1
    print(x)

0
1
2
3
4
11
12
13
14
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63
65
67
69
71
73
75
77
79
81
83
85
87
89
91
93
95
97
99
天安门
东方明珠
珠江
1
2
3
4
5
6


In [39]:
# 注：这个''代表空字符串
# pwd = ''  
# while pwd != '520666':
#     pwd = input('请输入银行卡密码：')
# print('卡内还有999999999999元~')

a=0
while a<4:
    a = a+1
print(a)

a = 0
while a < 4:
    a = a + 1
    print("while循环内")
    print(a)
print("while循环外")
print(a)

for i in range(1,11):
    if i!=4:
        print(i)
        
#         height = 0   
# while height<170:
#     height = int(input('继续输入身高：'))
# print('已经找到人去搬水了~')

4
while循环内
1
while循环内
2
while循环内
3
while循环内
4
while循环外
4
1
2
3
5
6
7
8
9
10


In [50]:
print(3<6)
print(3>6)
print('北京'=='北京')  
print('北京'=='首都')

import time
# time.sleep(1)

# 请阅读代码后再直接运行


print('以下数据判断结果都是【真】：')
print(bool(True))
print(bool(10))
print(bool('xyz'))

print('以下数据判断结果都是【假】：')
print(bool(False))
print(bool(0))
print(bool(''))
print(bool(None))

# 请先阅读代码，然后直接运行
a = 5
b = -5

print('以下是and运算')
if a==5 and b==5:    # 【b实际上是-5】
    print('True')
else:
    print('False')

print('以下是or运算')
if a==5 or b==5:  # 【b实际上是-5】
    print('True')
else:
    print('False')
    
    # 请先阅读代码，然后直接运行
list = [1,2,3,4]

a = 1
b = 10
c = 0
print("in 计算")
print(bool(a in list))
print(bool(b in list))
print(bool(c in list))

# 请先阅读代码，然后直接运行
for i in range(1,10):
    if(i>5):
        break
    print(i)

x = 0
while x<10:
    if(x>4):
        break
    x=x+1
    print(x)
    
    # 请先阅读代码，然后直接运行
for i in range(1,11):
    if i==5:
        continue
    print(i)

x = 0
while x<10:
    x=x+1
    if x==5:
        continue
    else:   
        print(x)

True
False
True
False
以下数据判断结果都是【真】：
True
True
True
以下数据判断结果都是【假】：
False
False
False
False
以下是and运算
False
以下是or运算
True
in 计算
True
False
False
1
2
3
4
5
1
2
3
4
5
1
2
3
4
6
7
8
9
10
1
2
3
4
6
7
8
9
10


In [52]:
# 请先阅读代码，然后直接运行
age = int(input('请输入你的年龄:'))
if age >= 18:
    pass
else:
    print('你未成年，不得进入网吧')

请输入你的年龄:1
你未成年，不得进入网吧


In [59]:
#else语句不仅能够在if语句中使用，还能够和循环搭配使用
for i in range(3):
    num = int(input('请输入0结束循环，你有3次机会:'))
    if num == 0:
        print('你触发了break语句，导致else语句不会生效。')    
        break
else:
    print('3次循环你都错过了，else语句生效了。')
    
    
# 请先阅读代码，然后直接运行
num=0
while num<3:
    n = int(input('请输入0结束循环，你有3次机会:'))
    if n == 0:
        print('你触发了break语句，导致else语句不会生效。')    
        break
    num+=1
else:
    print('3次循环你都错过了，else语句生效了。')

请输入0结束循环，你有3次机会:0
你触发了break语句，导致else语句不会生效。
请输入0结束循环，你有3次机会:0
你触发了break语句，导致else语句不会生效。


In [61]:
num = 7  #设定秘密数字
while True:       
    guess = int(input('你们猜猜数字是多少:'))   #输入猜测数字
    if guess==num:  #数字对比
        print('正确！给你一个苹果') 
        break
    elif guess>num:
        print('你猜的太大了，重新猜猜~')
    else:
        print('你猜的太小了，重新猜猜~')

你们猜猜数字是多少:7
正确！给你一个苹果


In [62]:
for i in range(1,21):
    if i%3  == 0:
        continue
    else:
        print(i)

1
2
4
5
7
8
10
11
13
14
16
17
19
20
